restli-tools/src/main/java/com/linkedin/restli/tools/snapshot/check/PegasusSchemaSnapshotCompatibilityChecker.java

```diff
@@ -266,7 +268,19 @@ public CompatibilityInfoMap checkPegasusSchemaCompatibility(String prevPegasusSc
 
     if (!result.getMessages().isEmpty())
     {
-      result.getMessages().forEach(message -> _infoMap.addModelInfo(message));
+      if (_isExtension)
```

Comment:
A cleaner solution will be to add a new Mode `EXTENSION` in CompatibilityOptions::Mode and use that to change the behavior of CompatibilityChecker

Reply:
Agree that additional mode will be better than an additional flag. we should keep flags as few as possible.